libsolidity/analysis/DeclarationContainer.cpp

```diff
@@ -42,13 +43,29 @@ Declaration const* DeclarationContainer::conflictingDeclaration(
 	if (m_invisibleDeclarations.count(*_name))
 		declarations += m_invisibleDeclarations.at(*_name);
 
+	// Since functions can be overloaded even within a single contract, `_inheriting` is not checked.
 	if (dynamic_cast<FunctionDefinition const*>(&_declaration))
 	{
 		// check that all other declarations with the same name are functions
 		for (Declaration const* declaration: declarations)
 			if (!dynamic_cast<FunctionDefinition const*>(declaration))
 				return declaration;
 	}
+	// Until we implement #1245, there are no special cases for events
```

Comment:
@chriseth I added checkboxes in https://github.com/ethereum/solidity/pull/1245#issuecomment-263846865